zipline/pipeline/loaders/utils.py

```diff
@@ -112,11 +114,15 @@ def previous_event_indexer(all_dates,
         ``event_{dates,timestamps,sids}``.
     """
     validate_event_metadata(event_dates, event_timestamps, event_sids)
-    out = np.full((len(all_dates), len(all_sids)), -1, dtype=np.int64)
+    out = np.full(
+        (len(data_query_cutoff_times), len(all_sids)),
+        -1,
+        dtype=np.int64,
+    )
 
     eff_dts = np.maximum(event_dates, event_timestamps)
     sid_ixs = all_sids.searchsorted(event_sids)
-    dt_ixs = all_dates.searchsorted(eff_dts)
+    dt_ixs = data_query_cutoff_times.searchsorted(eff_dts, side='right')
```

Comment:
Reasoning through this change:

`data_query_cutoff_times` is the index into which we're aligning our input data. For each event record `i`, `dt_ix[i]` should be the index into the cutoff times of the first date on which we should show the event's data. The only case where `searchsorted` produces different values is when the value being looked up appears in the searched data. so this is effectively saying that if an event has a timestamp that's exactly on the data_query_cutoff_time, we push it forward to the next day.

Did I get that right? If so, what prompted this change? This seems like a semi-arbitrary distinction.

Reply:
we use side=right in the blaze core loader. I wanted to consistently handle this case wherever we care about the cutoff time.